src/ui/select/select.tsx

```diff
@@ -44,28 +47,36 @@ export const Select = ({
   }, [])
 
   const toggleSelection = (option: SelectOption) => {
-    onSelect(option)
     if (multiSelect) {
       setSelected((prevSelected) => {
         if (prevSelected.includes(option)) {
-          return prevSelected.filter((selectedOption) => selectedOption !== option)
+          const newValue = prevSelected.filter((selectedOption) => selectedOption !== option)
+          onSelect(newValue)
+          return newValue
         } else {
-          return [...prevSelected, option]
+          const newValue = [...prevSelected, option]
+          onSelect(newValue)
+          return newValue
         }
       })
     } else {
+      onSelect([option])
       setSelected([option])
     }
   }
 
   const isSelected = (option: SelectOption) => selected.includes(option)
 
+  const visiblePlaceholder =
+    (selected.length ? shortenString(selected.join(', '), 30) : null) ?? placeholder ?? name
```

Comment:
Why do you need `null` here?

```suggestion
  const visiblePlaceholder =
    (selected.length ? shortenString(selected.join(', '), 30) : placeholder) ?? name
```